optimum/neuron/utils/input_generators.py

```diff
@@ -73,3 +76,93 @@ def generate(self, input_name: str, framework: str = "pt", int_dtype: str = "int
+        "mid_block_additional_residual",
+    )
+
+    def __init__(
+        self,
+        task: str,
+        normalized_config: NormalizedTextConfig,
+        batch_size: int,
+        sequence_length: Optional[int] = None,
+        num_channels: Optional[int] = None,
+        height: Optional[int] = None,
+        width: Optional[int] = None,
+        vae_scale_factor: Optional[int] = None,
+        encoder_hidden_size: Optional[int] = None,
+        **kwargs,
+    ):
+        self.task = task
+        self.normalized_config = normalized_config
+        self.batch_size = batch_size
+        self.sequence_length = sequence_length
+        self.num_channels = num_channels
+        self.height = height
+        self.width = width
+        self.vae_scale_factor = vae_scale_factor
+        self.text_encoder_hidden_size = encoder_hidden_size
+
+    def generate(self, input_name: str, framework: str = "pt", int_dtype: str = "int64", float_dtype: str = "fp32"):
+        if input_name == "timestep":
+            shape = [self.batch_size]
+            return self.random_int_tensor(shape, max_value=999, framework=framework, dtype=int_dtype)
+        elif input_name == "encoder_hidden_states":
+            shape = (self.batch_size, self.sequence_length, self.text_encoder_hidden_size)
+            return self.random_float_tensor(shape, framework=framework, dtype=float_dtype)
+        elif input_name == "controlnet_cond":
+            num_channels = getattr(
+                self.normalized_config, "conditioning_channels", 3
+            )  # num_channels = 3 since `do_convert_rgb=True`
+            shape = (
+                self.batch_size,
+                num_channels,
+                self.height * self.vae_scale_factor,
+                self.width * self.vae_scale_factor,
+            )
+            return self.random_float_tensor(shape, framework=framework, dtype=float_dtype)
+        elif input_name == "conditioning_scale":
+            return torch.tensor([1.0])
+        elif input_name == "down_block_additional_residuals":
+            sample_shape = (self.batch_size, self.normalized_config.block_out_channels[0], self.height, self.width)
+            sample = self.random_float_tensor(sample_shape, framework=framework, dtype=float_dtype)
+            down_block_res_samples = (sample,)
+            num_past_cross_attn_blocks = 0
+            height = copy.deepcopy(self.height)
+            width = copy.deepcopy(self.width)
+            for idx, down_block_type in enumerate(self.normalized_config.down_block_types):
+                res_samples = ()
+                shape = (self.batch_size, self.normalized_config.block_out_channels[idx], height, width)
+                for _ in range(self.normalized_config.layers_per_block):
+                    res_samples += (self.random_float_tensor(shape, framework=framework, dtype=float_dtype),)
+                if idx == len(self.normalized_config.down_block_types) - 1:
+                    pass
```

Comment:
Is it wanted? Otherwise I would just change the condition to:
```
idx != len(self.normalized_config.down_block_types) - 1
```
And put the body of the `else` branch inside.